### DISCUSSION TITLE
Post SEO options in head.hbs

### DISCUSSION THREAD

**Discussion by yowalsh** [GetPublii/Publii#1539] [Help]
I've tried and tried, and am looking to add a custom code to the head.hbs. I've set up a field under OTHER in Post Options and want it to display in the header of all pages.

Example: meta name="keywords" content="{{@config.post.displayKeywords}}" 

I have this which results in a new textarea field under OTHER:

        {
            "name": "displayKeywords",
            "label": "Page Keywords",
            "note": "Enter page keywords here",
            "type": "textarea"
        },

I can enter and save keywords in the textarea.... but not showing in the header.  Any ideas?  Thanks!

**Comment by candidexmedia** [comment]
What have you added to your `head.hbs` partial code so far?

You've added the extra Post settings options, which is perfect, but now it has to be referenced in your partial, and it's not clear if you've done that already.

It would look something like this, close to the other meta stuff in `head.hbs`:

```hbs
...
<meta meta name="keywords" content="{{ @config.post.displayKeywords }}">
...
```

See: [How to access option values of the post settings in the theme?](https://getpublii.com/dev/theme-settings-api/#howtoaccessoptionvaluesofthepostsettingsinthetheme)

**Reply by yowalsh** [reply]
UPDATE: using this code {{ @config.post.displayKeywords }} displays the keywords for all non-frontpage posts, so all set there.

PENDING: Currently I have the frontpage displaying a specific post. The keywords do **not** display on the homepage. The function above does not seem to work on the homepage. How can this be updated?

QUESTION: I currently added these to the OTHER section in the post editor. Is there a way to add fields to show in the SEO section (which would be more logical in this case?)

Thanks!

=============

Thank you for your reply. That is what I have going in the head.hbs, but it is not loading the keywords up. Not on front-page, nor on any post page.

See images attached - my code in head.hbs, config.json, and the result in the <head>

Thank you so much!
![config-json](https://github.com/GetPublii/Publii/assets/26417741/599ef174-13d5-4ff1-9b6e-f88885fcc238)
![head hbs](https://github.com/GetPublii/Publii/assets/26417741/7dccf22e-229e-4e4d-9c8b-dc06e5cb34c0)
![html-jpg](https://github.com/GetPublii/Publii/assets/26417741/fb9bdabe-8f74-4eb0-af6e-9341ab2b1804)

**Reply by candidexmedia** [reply]
> UPDATE: using this code {{ @config.post.displayKeywords }} displays the keywords for all non-frontpage posts, so all set there.

Exactly! You were just missing spaces after and before the curly brackets

> PENDING: Currently I have the frontpage displaying a specific post. The keywords do not display on the homepage. The function above does not seem to work on the homepage. How can this be updated?

I haven't tested on my end, but I think using postViewConfig might be the solution.

This is explained in [How to access option values of the post settings in the theme?](https://getpublii.com/dev/theme-settings-api/#howtoaccessoptionvaluesofthepostsettingsinthetheme): 

"Warning! The @config.post.* syntax works only the single post templates. Under pages with post listings you have to use the postViewConfig object."

"If you want to get the same value under posts listing (e.g.: <mark>on the frontpage</mark>) you should use:

`{{ postViewConfig.headerColor }}`"

In your case that would be:

```hbs
...
<meta meta name="keywords" content="{{ @postViewConfig.displayKeywords }}">
...
```

That said, not 100% sure, and would defer to the devs on that one since frontpage posts might require their own context/logic or some additionnal handlebars. I'm also not sure if this would be an `index.hbs` modification, or `head.hbs` partial mod...

> QUESTION: I currently added these to the OTHER section in the post editor. Is there a way to add fields to show in the SEO section (which would be more logical in this case?)

AFAIK: No. Custom post settings go in "Other", but I'll let the devs have the final say on that one. Might be a good Feature Request to create in the Issues tab of this Github repo.
